python/cuml/dask/common/utils.py

```diff
@@ -162,3 +162,44 @@ def serialize_mat_descriptor(m):
 
     patch_func()
     client.run(patch_func)
+
+from threading import Lock
+
+class ConcurrentPartitionLock:
```

Comment:
JFYI- I still need to add a timeout to this, but figured it would be useful to push what I've done so far so that it can be tested by other folks.